orientjs/orientjs.d.ts

```diff
@@ -0,0 +1,500 @@
+
+        interface Cluster {
+            name: string;
+            location: string;
+            list(refresh?: boolean): Promise<any[]>;
+            create(name: string, location?: string): Promise<Cluster> & Promise<any>;
+            get(nameOrId: string, refresh?: boolean): Promise<Cluster> & Promise<any>;
+            get(nameOrId: number, refresh?: boolean): Promise<Cluster> & Promise<any>;
+            getByName(name: string, refresh?: boolean): Promise<Cluster> & Promise<any>;
+            getById(id: string, refresh?: boolean): Promise<Cluster> & Promise<any>;
+            drop(name: string): Promise<Db>;
+            count(name: string): Promise<Number>;
+            range(name: string): Promise<any>;
+            cacheData(clusters: Cluster[] & any[]): Db;
+
+        }
+
+        interface RecordMeta {
+            "@rid": string|RID;
+            "@version": string|number;
+           
+        }
+
+        interface Record {
+            rid: RID;
+            create(record: any, options?: any): Promise<Record> ;
+            get(record: Record & any, options?: any): Promise<Record | Buffer>;
+            resolveReferences(records: Record[] &  any[]): any;
+            meta(record: Record & any, options?: any): Promise<RecordMeta> ;
+            update(): Promise<Record> & Promise<any>;
+            update(record: Record & any, options?: any): Promise<Record> ;
+            delete(): Promise<Record> & Promise<any>;
+            delete(record: Record & any, options?: any): Promise<Record> ;
+            delete(record: RID, options?: any): Promise<Record> ;
+            delete(record: string, options?: any): Promise<Record> ;
+
+        }
+
+        interface Index {
+            cached: boolean;
+            db: Db;
+            name: string;
+            algorithm: string;
+            clusters: Cluster[];
+            type: string;
+
+            configure(config: any): void;
+            add(args: any): Promise<Index[]>;
+            add(args: any[]): Promise<Index[]>;
+            get(key: string): Promise<RID>;
+            set(key: string, value: string): Promise<Index>;
+            set(key: string, value: RID): Promise<Index>;
+            delete(name: string): Promise<Index>;
+            select(): Statement;
+            list(refresh?: boolean): Promise<Index[]>;
+            create(config: any): Promise<Index>;
+            drop(name: string): Promise<Db>;
+            get(name: string, refresh?: boolean): Promise<Index>;
+            cacheData(indices: any[]): Promise<Db>;
+
+        }
+
+        interface Statement extends Query<any> {
+            select(param: string): Statement;
+            select(params: string[]): Statement;
+            traverse(param: string): Statement;
+            traverse(param: string[]): Statement;
+
+            strategy(param: string): Statement;
+            insert(param: string): Statement;
+            insert(param: string[]): Statement;
+            insert(param: any): Statement;
```

Comment:
same as my comment about `any` earlier. there is no reason to list these overloads if there is an `any` overload. just keep the `any` overload.